### DISCUSSION TITLE
Происходит сброс соединения

### DISCUSSION THREAD

**Discussion by bulsond** [hufrea/byedpi#66] [General]
Вроде настроил как положено. Запускаю строкой   `ciadpi -i 127.0.0.1 -I 192.168.100.207 --fake -1 --ttl 8` , в браузере прописал  сокс прокси 127.0.0.1 с портом 1080, и когда захожу на ютуб в консоле сыплет сообщениями `recv: Connection reset by peer`.  И видео понятное дело не загружаются.

**Comment by lufog** [comment]
Смахивает на слишком большой ttl пакета (fake добирается до сервера). Попробуй его уменьшить, ну или попробуй добавить `--disorder 1`. Поведение которого на Windows, позволяет не запариваться с ttl.

```
TTL необходимо подбирать такой, чтобы пакет прошел через все DPI, но не дошел до сервера.  

...

Для Windows есть еще один способ избежать обработки фейка сервером.
Это комбинирование fake с disorder:
- Параметры: --disorder 1 --fake 7
- Порядок отправки: 2-7 фейк, 7-30 оригинал, 1-30 оригинал

Если поддельный пакет и дойдет до сервера, то он будет перезаписан из-за полной ретрансмисси.
```

**Comment by bulsond** [comment]
У меня убунта, а не винда.
 Пробовал запускать так:
 `ciadpi -i 127.0.0.1 -I 192.168.100.207 --disorder 1 --auto=torst --tlsrec 1+s`  и так
 `ciadpi -i 127.0.0.1 -I 192.168.100.207 --disorder 1 --fake -1 --md5sig`.
  Сообщения о сбросе соединения пропали. Но видео на ютубе не загружается. Причем я пробовал ваш проект. Потом SpoofDPI и он тоже не работает :(

**Reply by o-pisetsky** [reply]
Тоже сижу на убунте. Использую SmartProxy расширение, создаёшь новый прокси 127.0.0.1 и порт 1080 (localhost, а 1080 дефолт порт ciadpi), а потом создаёшь умный профиль и суёшь туда домен ютуба. Выбираешь свой кастомный прокси, запускаешь ./ciadpi -s 2 -d 2 и готово)

**Reply by o-pisetsky** [reply]
У меня выключен QUIC и Kyber! Юзаю мозиллу

**Comment by lufog** [comment]
Попробуй ради интереса просто `--oob 1`, на моем операторе, он отлично пробивается к ютубу. А в целом, тут нужно садится перебирать разные параметры, экспериментировать, и пытаться вникать. Рабочая комбинация есть практически в 99% случаев, но нужно приложить усилия чтобы ее найти.

**Comment by bulsond** [comment]
Пробовал так: `ciadpi -i 127.0.0.1 -I 192.168.100.207 --disorder 1 --fake -1 --md5sig --oob 1`.
 Стало сыпать такими сообщениями:
 ```shell
split cancel: pos=672-1, n=673
split cancel: pos=1245-1, n=1246
split cancel: pos=662-1, n=663
split cancel: pos=434-1, n=435
split cancel: pos=1253-1, n=1254
split cancel: pos=667-1, n=668
split cancel: pos=658-1, n=659
split cancel: pos=658-1, n=659
split cancel: pos=686-1, n=687
split cancel: pos=686-1, n=687
```
Видео не грузится все равно.

**Comment by lufog** [comment]
А просто `ciadpi -i 127.0.0.1 -I 192.168.100.207 -oob 1+s` не пробовал?

Сообщение типа `split ... pos ...`, у меня тоже появляются с `--disorder`. Насколько я понимаю, они просто информационные, но это не точно)

**Comment by bulsond** [comment]
Стало интереснее. Сообщения о сплите пропали. Ютуб перестал грузится. После перезапуска браузера та же фигня, как будто первые кадры появляются, а потом опять колесо загрузки...

**Comment by lufog** [comment]
В общем тут только экспериментировать. 

Можно конечно попробовать что-то из программ/скриптов, помогающих подобрать методы обхода. Но я знаю только о: 
- blockcheck.sh из [bol-van/zapret](https://github.com/bol-van/zapret), но сам не пробовал.
- DPI Tunnel (на Android), в нем, в профилях будет кнопка подобрать настройки, в заблокированном ресурсе укажи googlevideo.com например. Глянь что он подбирает у себя, и укажи аналогичные параметры тут.

**Comment by hufrea** [comment]
> split cancel: pos=672-1, n=673

Позиции нужно указывать в порядке возрастания, иначе разбиение отменяется.

У вас QUIC отключен?

**Comment by bulsond** [comment]
Я пробовал на Firefox и включал и выключал, разницы нет. Оставил выключен.

**Comment by z18d7** [comment]
Перепробовал много параметров и, к своему удивлению, обнаружил, что лучшие результаты достигаются с одним дефолтным параметром `--disorder 1`. При использовании `--fake` видео либо не грузится, либо тормозит.

Проводной билайн, Manjaro

**Comment by vladiscripts** [comment]
Подтверждаю. Скачал `git clone ...`, скомпилировал `make`, запустил `ciadpi  --disorder 1`. В браузере в расширении задал прокси как socks 127.0.0.1:1080. 
Заработал Youtube, стали открываться сайты торрентов (кроме kinozal.tv, хотя вроде через раз открывается... он и с Антизапрет/vpn работает нестабильно).
Ростелеком, Ubuntu.

**Comment by halimonenko** [comment]
> Я пробовал на Firefox и включал и выключал, разницы нет. Оставил выключен.

Сейчас Firefox не поддерживает проксирование UDP и QUIC (через прокси т.е.). Разработчики утверждают что это в дорожной карте на будущее. Поэтому для Firefox ничего не нужно выключать, где можно и быстрее (у него есть режим гонки) - он будет использовать HTTP3, где нельзя - не будет. Ничего можно не трогать и когда UDP проксирование заработает.

**Comment by exorevan** [comment]
Происходит то же самое с Android версией. Что там можно поменять в настройках для избежания таких ситуаций?

**Comment by finn1199** [comment]
Я так настроил в Arch Linux + Openbox:
1. Скачал отсюда тар-болл **byedpi-12-x86_64.tar.gz**
2. Распаковал его в созданную в Домашнем каталоге папку **byedpi**
3. Переименовал распакованный файл **ciadpi-12-x86_64** в просто **ciadpi**
4. В **Firefox** -> **Настройки** -> **Основные** -> **Настройки сети** -> **Настроить**
- здесь отмечаем **радио**-кнопкой пункт **Ручная настройка прокси**
- в поле/секции **Узел SOCKS** вписываем:
    **IP**-адрес `127.0.0.1`
    а в поле **Порт** `1080`
- по умолчанию под полем/секцией **Узел SOCKS** отмечена **радио**-кнопка **SOCKS5**, оставляем её (если отмечена другая **SOCKS4**, то отмечаем именно **SOCKS5**!!!) и жмём на кнопку **ОК**
5. В файле` /home/user/.config/openbox/autostart` добавляем запись:
(вместо **user** ваше имя **Пользователя** в системе)
`(sleep Хs && byedpi/ciadpi -i 127.0.0.1 -p 1080 -s 2) &`
или
`(sleep Хs && byedpi/ciadpi -i 127.0.0.1 -p 1080 -d 1) &`
(вместо **Xs** пишите количество секунд задержки очерёдности загрузки например **5s** (у меня по очерёдности например этот процесс 13-м в очереди и у меня записано 13s), параметр **-s** - можете поэкспериментировать со значением меньше/больше и смотрите, как будет грузиться **Youtube**. Здесь есть описание этого параметра)

По **п. 1**, если это **Arch**-чевидный дистр, можно не скачивать **тар**-болл, а просто из **AUR** установить пакет **byedpi-bin 0.12-1**:
`yay -S byedpi-bin`
И тогда в **п. 5** команда в автозапуск будет такой:
`(sleep Хs && ciadpi -i 127.0.0.1 -p 1080 -s 2) &`
или
`(sleep Хs && ciadpi -i 127.0.0.1 -p 1080 -d 1) &`
(из опыта, работа этого пакета из **AUR** не стабильна, хотя версия та же, иногда на пару секунд подвисает **Ютуб**.)

Руками можно запустить в **Терминале** на текущую сессию так:
`byedpi/ciadpi -i 127.0.0.1 -p 1080 -s 2 &`
или
`byedpi/ciadpi -i 127.0.0.1 -p 1080 -d 1 &`

А если установка была из **AUR** то:
`ciadpi -i 127.0.0.1 -p 1080 -s 2 &`
или
`ciadpi -i 127.0.0.1 -p 1080 -d 1 &`

P.S. для других **DE** или дистров **Linux** на **Systemd** можно и **юнит** запуска для **Systemd** сделать и запустить его...

Перезагружаем систему и...
Радуемся жизни...

Открываются практически все блоки РКН, кроме FB.
Х открывается, но медиа очень долго грузит...
Скорость, как была от прова РТ 800 Мб, так и есть!

**Дополнение:**
Потестил ещё разные варианты запуска, и вот на пока что оптимальный вариант для меня такой:
`byedpi/ciadpi -i 127.0.0.1 -p 1080 -s 2 -f -1 --md5sig`
Х открывается на УРА, FB так и не алё...

Оказалось, что при таком варианте Госуслуги тоже не открываются, начал снова экспериментировать с параметрами и вот в таком варианте Госы стали открываться:
`byedpi/ciadpi -i 127.0.0.1 -p 1080 -s 2 -M h,d -A torst -T 3 -f -1 -r 3+s -S`
FB и Instagram так и не алё...

**Reply by katefromr** [reply]
От всей души благодарю Вас! Задалась целью настроить и на компьютере (с телефоном разобралась). Моих знаний недостаточно и из-за этого поначалу психовала и хотела бросить, но упёртости хватило:)

У меня набор такой: система **Manjaro**, браузер **Mozilla Firefox**, провайдер **Yota**. 
Сделала по Вашей инструкции первые 4 пункта. В 5 не поняла, видимо, в Manjaro как-то иначе это делается, в общем я пропустила это (если можете подсказать, буду благодарна).
Из папки с файлом ciadpi запустила терминал и написала команду `./ciadpi -i 127.0.0.1 -p 1080 -s 2 -f -1 --md5sig`

Ютюб отлично грузит, счастью нет предела!
Спасибо, что делитесь опытом!

**Comment by bulsond** [comment]
Я рад за вас. А что за провайдер РТ? У меня дом.ру. Остальное, что вы написали по содержанию ничего нового мне не дает.

**Comment by finn1199** [comment]
> РТ

Ростелеком Северо-Запад.

> по содержанию ничего нового мне не дает

А я не для давания нового, а просто, люди многие мытарят, вопрошают, кричат, что не работает, а как, как же завести всё это дело, и вот, чтобы было понятно людам, что всё работает и поделился личным опытом...

**Comment by Hobbix** [comment]
Подскажите. Если запускаю **goodbyedpi** с дефолтным параметром "goodbyedpi.exe -9 --blacklist ..\russia-blacklist.txt --blacklist ..\russia-youtube.txt", то всё работает. Блокировки обходит, ютуб грузится нормально.

А вот если запускаю **byedpi** с дефолтным параметром "ciadpi.exe --ip 192.168.1.10 --port 9090 --split 1+s --disorder 3+s --mod-http=h,d --auto --tlsrec 1+s", то ничего уже не работает. Вернее обычные сайты по-прежнему грузятся, но заблокированные сайты не разблокирует и замедления ютуба не обходит. Сокс прокси byedpi прописываю в Firefox.

Подскажите правильный параметр для byedpi, который будет делать тоже самое, что и "goodbyedpi.exe -9 --blacklist ..\russia-blacklist.txt --blacklist ..\russia-youtube.txt".
Нужно именно решение с прокси для браузера от byedpi. Не хочу использовать goodbyedpi для всей системы сразу.

**Comment by halimonenko** [comment]
ciadpi-x86_64 -i 127.0.0.1 -p 1080 -d 1
т.е. -d 1

**Comment by halimonenko** [comment]
не предназначен для разблокировок сайтов, только для обхода замедления - надежно, быстро, безопасно

**Comment by Hobbix** [comment]
> ciadpi-x86_64 -i 127.0.0.1 -p 1080 -d 1
> т.е. -d 1

Странно, у меня почему-то это не работает, ютуб всё равно жутко тормозит. Пробовал и включать и отключать опцию network.http.http3.enable в браузере, но она похоже ни на что не влияет. Как только запускаю "goodbyedpi.exe -9", то сразу ютуб оживает. Получается параметра "-d 1" недостаточно? Может это как-то связано с тем, что запускаю не как службу, а просто батником (ярлыком с параметром)?

**Comment by halimonenko** [comment]
отключать http3 не нужно firefox пока не направляет его на прокси, а точно трафик идет через проксю byedpi? вы в браузере прокси указали?
и конечно все другие фильтры нужно временно убрать в том числе goodbydpi

**Comment by Hobbix** [comment]
> отключать http3 не нужно, а точно трафик идет через проксю byedpi? вы в браузере прокси указали?
> и конечно все другие фильтры нужно временно убрать в том числе goodbydpi

Да, конечно, все другие программы отключил перед запуском byedpi. Прокси в браузере прописал и он точно работает. Если закрыть консоль с ciadpi.exe (погасить прокси), то сайты в браузере перестают грузится любые. Как только запускаю ciadpi.exe с проксей, то сайты начинают грузится, но вот скорость на ютубе режет по-прежнему.

**Comment by halimonenko** [comment]
Secure DNS/SNI включили? в браузере?

**Comment by Hobbix** [comment]
> Secure DNS/SNI включили? в браузере?

В настройках Firefox был включен DoH. 
Когда проверил работает оно или нет тут: https://www.cloudflare.com/ru-ru/ssl/encrypted-sni/
то оказалось, что защита DNS и SNI фактически не работала. В общем оказалось, что нужно снять галочку в настройках прокси в Firefox с пункта "Отправлять DNS-запросы через прокси при использовании SOCKS 5". Как только эту галочку снял, то на вышеуказанном сайте сразу же показало, что защита DNS и SNI у меня успешно заработала.
Тут я уже подумал, что близок к успеху и полез на ютуб проверять. К сожалению, это не дало результата и ютуб по-прежнему еле живой. То есть с замедлением byedpi не справился (с параметром "-d 1").

**Comment by halimonenko** [comment]
Если не секрет - провайдер, город, ОС?

**Comment by finn1199** [comment]
> То есть с замедлением byedpi не справился (с параметром "-d 1").

Вы пробовали, как у меня в посте варианты запуска?
У меня Северо-Западный Ростелеком, оптика 800Мб, всё работает.
Ютуб на УРА показывает в HD, не все, но многие сайты из блока РКН тоже открываются.
Кинозал, ННМ, Рутрекер...
X открывается, но медиа подгружает с запозданием.
FB не открывается.

**Comment by bulsond** [comment]
У меня Петербург Дом.ру. Перепробовал многое:
- [Green Tunnel](https://github.com/SadeghHayeri/GreenTunnel)
- [SpoofDPI](https://github.com/xvzc/SpoofDPI)
- [DPI Tunnel CLI](https://github.com/nomoresat/DPITunnel-cli)
- [ByeDPI](https://github.com/hufrea/byedpi).
В итоге сработало расширение для хрома `Youtube Unblocked`.
Да, еще работает через мобилку, если сделать её точкой доступа.

**Comment by Hobbix** [comment]
> Если не секрет - провайдер, город, ОС?

Win10, региональный провайдер. 
Вообще я думал, что byedpi это тот же goodbydpi только в виде прокси, но видимо это не так. Goodbydpi с параметром -9 работает с этим же провайдером успешно, ютуб оживляет. Уж не знаю, что там скрывается за опцией -9, но она реально работает. Уж очень мне хотелось бы добиться успеха с помощью этой прокси от bydpi.
Например, на android у меня успешно заработал PowerTunnel с опциями https "фрагментация пакетов - вкл" и  "размер фрагмента - 2". Не знаю, что они делают, но это тоже работает.

> Вы пробовали, как у меня в посте варианты запуска?

Перепробовал все комбинации параметров, которые указаны в этой теме.

**Comment by halimonenko** [comment]
> Перепробовал все комбинации параметров, которые указаны в этой теме.

скажите пожалуйста, у вас Windows 11 показал уведомление на предоставление разрешения для доступа к сети для ByeDPI при первом использовании?

**Comment by finn1199** [comment]
> Перепробовал все комбинации параметров, которые указаны в этой теме

Я правильно понимаю, Windows?
Порт открыт в Windows Firewall?
`netsh advfirewall firewall add rule name="ByeDPI protocol="TCP" localport=9090 action=allow dir=IN`
или вместо `protocol="TCP"` вписать `protocol="UDP"`

**Comment by finn1199** [comment]
> * сработало расширение для хрома

я вот не любитель Хрома, от слова совсем.
Лисичку пользую, поэтому под неё искал решение...

**Comment by lufog** [comment]
@finn1199 почему не указать в настройках Windows на всю систему? У меня вроде всеми программами (кроме Windows Store и UWP) прокси подхватывалось.
Settings -> Internet -> Proxy (может отличаться, пишу по памяти, плюс у меня Windows на английском)
IP `socks=127.0.0.1` port `1080`

**Comment by Hobbix** [comment]
> скажите пожалуйста, у вас Windows 11 показал уведомление на предоставление разрешения для доступа к сети для ByeDPI при первом использовании?

У меня eis и win10, все разрешения на входящий и исходящий для ciadpi.exe конечно даны. Впрочем я уже выше писал, что прокси точно работает. Проверяется легко. Если в браузере Firefox указано соединение через socks прокси, а прокси бы не работал (например из-за firewall или просто выключен), то у меня не открылся бы вообще ни один сайт. Так что firewall его точно не блокирует.
